osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -367,6 +367,8 @@ public void Edit(BeatmapInfo beatmapInfo = null)
                 throw new InvalidOperationException($"Attempted to edit when {nameof(AllowEditing)} is disabled");
 
             Beatmap.Value = beatmaps.GetWorkingBeatmap(beatmapInfo ?? beatmapInfoNoDebounce);
+            Ruleset.Value = beatmapInfo?.Ruleset;
```

Comment:
This will set the global game ruleset to null if `beatmapInfo` is not explicitly provided by the caller of this method. This may not lead to a crash, necessarily (see `TestSceneOsuGame.TestNullRulesetHandled()`), but it may fail to work correctly given the preceding line. If `Beatmap.Value` fell back to `beatmapInfoNoDebounce` then you will likely want to match that beatmap's ruleset, too.